nearcore/src/lib.rs

```diff
@@ -180,16 +180,11 @@ fn apply_store_migrations(path: &Path, near_config: &NearConfig) -> anyhow::Resu
         migrate_30_to_31(path, &near_config);
     }
 
-    #[cfg(feature = "nightly_protocol")]
-    {
+    if cfg!(feature = "nightly") || cfg!(feature = "nightly_protocol") {
```

Comment:
```suggestion
    if cfg!(feature = "nightly_protocol") {
```

nightly implies nightly_protocol, no?

Reply:
Yes, you’re correct.  There were two things I was thinking:
- Firstly I wanted to make extra sure we take this path even if we break Cargo.toml and remove the implication.
- Secondly, I wanted to avoid nightly_protocol checks in the code so that people aren’t steered towards using ‘--features nightly_protocol’ if they aren’t familiar with how the features work.  Part of the confusion with the flags was that some tests would check ‘cfg!(feature = "nightly_protocol")’ but then when you run them with ‘--features nightly_protocol’ they would fail because really they required ‘--features nightly’.

I don’t feel particularly strongly about either of the arguments though so I’m happy to update if you think it’s better.

Reply:
First point is rather convincing!